sql/src/main/java/org/apache/druid/sql/calcite/rel/Windowing.java

```diff
@@ -358,12 +359,43 @@ public WindowFrame getWindowFrame()
       if (group.lowerBound.isUnbounded() && group.upperBound.isUnbounded()) {
         return WindowFrame.unbounded();
       }
+      int lowerOffset = figureOutOffset(group.lowerBound);
+      int upperOffset = figureOutOffset(group.upperBound);
+      if (group.lowerBound.isPreceding()) {
+        lowerOffset = -lowerOffset;
+      }
```

Comment:
instead of adding `if`-s for both `lower`/`upper` - can we make these changes inside `figureOutOffset` ?